R/processCapabilityStudies.R

```diff
@@ -2387,5 +2394,15 @@ processCapabilityStudies <- function(jaspResults, dataset, options) {
   return(plotList)
 }
 
-
-
+.pcTableFormatNumbers <- function(number) {
+  if (all(is.na(number)) || all(is.null(number))) {
+    return(rep(NA, length(number)))
+  }
+  output <- formatC(number, format = "f", digits = .numDecimals)
+  output <- sub("\\.?0+$", "", output)
+  if (any(is.na(number)) || any(is.null(number))) {
+    naNumbers <- which(is.na(number) | is.null(number))
+    output[naNumbers] <- NA
+  }
+  return(output)
+}
```

Comment:
`any(is.null(number))` looks a bit odd because `c(NULL, NULL)` is just `NULL`. Also, since `NULL` has length 0 this returns `logical(0)` for NULL input. Maybe
```suggestion
.pcTableFormatNumbers <- function(number) {

  if (is.null(number))
    return(number)

  if (all(is.na(number)))
    return(rep(NA, length(number)))
  
  output <- formatC(number, format = "f", digits = .numDecimals)
  output <- sub("\\.?0+$", "", output)
  if (anyNA(number))
    output[is.na(number)] <- NA

  return(output)
}
```

Reply:
@vandenman Yes, that makes sense. Thanks!